MzML/MzmlMethods.cs

```diff
@@ -140,34 +325,47 @@ public static void CreateAndWriteMyMzmlWithCalibratedSpectra(IMsDataFile<IMsData
             // Loop over all spectra
             for (int i = 1; i <= myMsDataFile.NumSpectra; i++)
             {
+
                 mzML.run.spectrumList.spectrum[i - 1] = new Generated.SpectrumType()
                 {
-                    defaultArrayLength = myMsDataFile.GetOneBasedScan(i).MassSpectrum.Size,
-
-                    index = i.ToString(CultureInfo.InvariantCulture),
-                    id = myMsDataFile.GetOneBasedScan(i).OneBasedScanNumber.ToString(),
-
-                    cvParam = new Generated.CVParamType[8]
+                    defaultArrayLength = myMsDataFile.GetOneBasedScan(i).MassSpectrum.YArray.Length,
+                    index = (i - 1).ToString(CultureInfo.InvariantCulture),
+                    id = "controllerType=0 controllerNumber=1 " + "scan=" + (myMsDataFile.GetOneBasedScan(i).OneBasedScanNumber).ToString(),
```

Comment:
This format was taken from other mzML files that all seemed to specify the controllerType and Number in every spectrum, although it is not needed. The minimum required would look something like "scan=(index)"